zerver/forms.py

```diff
@@ -227,6 +227,25 @@ def clean_password(self) -> str:
         return password
 
 
+class DemoRegistrationForm(HowFoundZulipFormMixin, forms.Form):
```

Comment:
I didn't use the `RealmDetailsForm` here because of the `realm_subdomain` field.

There is a chance that we might want to send an initial subdomain option as an initial value on the form and let folks click update or something until they get a demo name that they like. If so, then we can likely implement that by updating this to use the `RealmDetailsForm` (with a modified clean method for the field).

If we don't end up adding that feature, then I'd likely deduplicate here by pulling out the subdomain field itself to a mixin form that's used in the `RegistrationForm` and `RealmCreationForm`, and make the `RealmDetailsForm` have only the three fields shared by all three forms: realm name, realm type and realm language.